File: apps/dashboard/types/api-key.ts
export interface APIKey {
  id: string;
  org_id: string;
  app_id: string;
  name: string;
  key_prefix: string;
  is_revoked: boolean;
  created_at: string;
  revoked_at: string | null;
}

export interface CreateAPIKeyRequest {
  name: string;
}

/**
 * Response from POST /v1/publisher-apps/:id/api-keys.
 * The `key` field contains the full plaintext — returned ONCE only.
 */
export interface CreateAPIKeyResponse {
  id: string;
  org_id: string;
  app_id: string;
  name: string;
  key: string;
  key_prefix: string;
  is_revoked: boolean;
  created_at: string;
}

export interface APIKeyListResponse {
  items: APIKey[];
}

File: services/api-dashboard/internal/model/api_key.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// APIKey represents a user's API key.
  type APIKey struct {
	ID        uuid.UUID  `json:"id"`
	OrgID     uuid.UUID  `json:"org_id"`
	AppID     uuid.UUID  `json:"app_id"`
	Name      string     `json:"name"`
	KeyHash   string     `json:"-"`
	KeyPrefix string     `json:"key_prefix"`
	IsRevoked bool       `json:"is_revoked"`
	CreatedAt time.Time  `json:"created_at"`
	RevokedAt *time.Time `json:"revoked_at"`
}
